### ISSUE TITLE
Add optional DBT package for Super-Linter for SQLFluff from v0.7.0

### ISSUE THREAD

**Issue by tunetheweb** [github/super-linter#2051]
Hi, I'm one of the maintainers of SQLFluff.

From our next release (0.7.0) we are splitting our some of the functionality into a separate, optional plugin that users can install only if they need that functionality.

The reason it's being split out is not all users need this, and it allows a separate release cadence, and to clean up our code, and to reduce the need for users who don't need it to install this code. There is no harm to install it even if it's not used.

Therefore I think the GitHub Super-Linter should include this by default. It won't be used unless referred to in the user's SQLFluff config file, but would need to be included in your Docker builds to allow them to continue using this functionality (which they may well have been happily using in the current version of the Super-Linter ). 

I've raised https://github.com/sqlfluff/sqlfluff/issues/1600 in the SQLFluff repo, as a reference for this to track adding this here so feel to ask any questions here or there.

The plugin is called `sqlfluff-templater-dbt` is currently part of the main SQLFluff repo, but is a separate pip package (the package is already available now as an alpha release (0.7.0a4): https://pypi.org/project/sqlfluff-templater-dbt/).

As it's an optional dependency it's not included in our Python `requirements.txt` dependence file so won't be installed by default based on how SQLFluff is currently pulled into the Super-Linter. How can we get this plugin added in as a dependency to the Super-Linter  at the same time as SQLFluff to ensure users can continue to avail of that functionality if they want it? As I said above, those who do not use DBT will not have any issues with this package being installed.

Happy to open a PR for this if you could give me some pointers of where to add it. Note this should only be added when the Super-Linter  upgrades to SQLFluff 0.7.0 which we expect to send live in next few days.

**Comment by admiralAwkbar** [comment]
@tunetheweb Awesome work on that
WOuld love to help and get that working here.
Let me know what I can do to assist :)

**Comment by tunetheweb** [comment]
Is it just a matter of adding this new package to the pipfile?: https://github.com/github/super-linter/blob/main/dependencies/Pipfile

Will that mean it's automatically installed in the Dockerfile along with sqlfluff?

Are there any other considerations I need to consider? Understand there's a light docker build? Do I need to worry about that?

Also probably should include a note in the documentation to say that plugin is included.

Anything else?

**Comment by tunetheweb** [comment]
Closing this. It appears Super Linter users do not use this functionality since heard no complaints. Can revisit if there is a desire to use this.

**Comment by 808sAndBR** [comment]
I would love to get this added, it looks like the PR got closed out without merge. Is there anything I can do to help get this live? I am getting the error of it not being available to find the templater (below). 

```
| Starting in sqlfluff version 0.7.0 the dbt templater is distributed as a seperate python package. Please pip install sqlfluff-templater-dbt to use it.
| Error loading config: Requested templater 'dbt' which is not currently available. Try one of raw, jinja, python
```

**Comment by 808sAndBR** [comment]
Actually, maybe this isn't as simple as just adding the `sqlfluff-templater-dbt` since we also need whatever connectors/sub-packages dbt needs outside of its core package (for me `dbt_snowflake`). 

I have also been poking around megalinter (https://github.com/megalinter/megalinter) which seems to be a hard fork of this project. They have a way to run "pre-commands" which is letting me pip install those packages, I do not think similar is possible in super-linter currently though. 

So I guess to really get this working we would either need a way to install additional dependencies, or pull in all the sub packages of both sql-fluff as well as dbt.

**Comment by tunetheweb** [comment]
Yeah I do think this might be better run outside of the super-linter to be honest so a project using dbt, as you say, likely has it's own Python env and requirements anyway. SQLFluff is the easiest install of that.

Which is basically where I landed for the equivalent PR for the mega-linter: https://github.com/megalinter/megalinter/pull/861#issuecomment-983015475, and also for our own Dockerfile maintained by SQLFluff: https://github.com/sqlfluff/sqlfluff/issues/2102#issuecomment-991973532

Where SQLFluff in Super-Linter makes more sense is for non-Python projects (e.g. T-SQL project), where the basic SQLFluff package suffices. So still think it's good to be in Super-Linter, but not sure we need the dbt-templater package as well. Which is why I closed this.
